webrender/src/frame_builder.rs

```diff
@@ -1098,16 +1080,45 @@ impl<'a> LayerRectCalculationAndCullingPass<'a> {
         }
     }
 
+    fn rebuild_clip_info_stack_if_necessary(&mut self, mut scroll_layer_index: ScrollLayerIndex) {
+        if let Some(previous_scroll_layer) = self.current_clip_stack_scroll_layer {
+            if previous_scroll_layer == scroll_layer_index {
+                return;
+            }
+        }
+
+        // TODO(mrobinson): If we notice that this process is expensive, we can special-case
+        // more common situations, such as moving from a child or a parent.
+        self.current_clip_stack_scroll_layer = Some(scroll_layer_index);
+        let mut new_clip_info_stack = Vec::new();
```

Comment:
Thanks! I could have sworn that didn't compile before. Likely this was a holdover from an older version of the code. I've fixed this in the latest version of the branch.